app/api/utils.py

```diff
@@ -242,45 +243,51 @@ class CoNLLParser(FileParser):
     ```
     """
     def parse(self, file):
-        words, tags = [], []
         data = []
         file = io.TextIOWrapper(file, encoding='utf-8')
-        for i, line in enumerate(file, start=1):
-            if len(data) >= settings.IMPORT_BATCH_SIZE:
-                yield data
-                data = []
-            line = line.strip()
-            if line:
-                try:
-                    word, tag = line.split('\t')
-                except ValueError:
-                    raise FileParseException(line_num=i, line=line)
-                words.append(word)
-                tags.append(tag)
-            elif words and tags:
-                j = self.calc_char_offset(words, tags)
+
+        # Add check exception
+
+        field_parsers = {
+            "ne": lambda line, i: conllu.parser.parse_nullable_value(line[i]),
+        }
+
+        gen_parser = conllu.parse_incr(
+            file,
+            fields=("form", "ne"),
+            field_parsers=field_parsers
+        )
+
+        try:
+            for sentence in gen_parser:
+                if not sentence:
+                    continue
+                if len(data) >= settings.IMPORT_BATCH_SIZE:
+                    yield data
+                    data = []
+                words, labels = [], []
+                for item in sentence:
+                    word = item.get("form")
+                    tag = item.get("ne")
+
+                    if tag is not None:
+                        char_left = sum(map(len, words)) + len(words)
+                        char_right = char_left + len(word)
+                        span = [char_left, char_right, tag]
+                        labels.append(span)
+
+                    words.append(word)
+
+                # Create JSONL
+                j = {'text': ' '.join(words), 'labels': labels}
```

Comment:
Nit: I'd inline this variable to avoid short/non-descriptive variable-names like `j`.